src/app/wallets/update-pending-invoices.ts

```diff
@@ -35,7 +35,14 @@ export const declineHeldInvoices = async (logger: Logger): Promise<void> => {
     logger,
     processor: async (walletInvoice: WalletInvoice, index: number) => {
       logger.trace("updating pending invoices %s in worker %d", index)
-      await declineHeldInvoice({ walletInvoice, logger })
+
+      walletInvoice.recipientWalletDescriptor.currency === WalletCurrency.Btc
+        ? await updatePendingInvoice({ walletInvoice, logger })
+        : await declineHeldInvoice({
+            pubkey: walletInvoice.pubkey,
+            paymentHash: walletInvoice.paymentHash,
+            logger,
+          })
```

Comment:
the difference in signature between `updatePendingInvoice` and `declineHeldInvoice` feels inadequate. but maybe it's ok?

Reply:
yes but in one case we pass `walletInvoice`, in the other case we pass `walletInvoice.pubkey` and `walletInvoice.paymentHash`, in so both case we have walletInvoice ?

feels to me that the if we were doing the destructuring of the walletInvoice inside declineHeldInvoice instead of here, then the signature function would be the same?

maybe what you're doing is that there are other place we are calling `declineHeldInvoice`?